interfaces/apparmor/spec.go

```diff
@@ -61,15 +67,30 @@ func (spec *Specification) AddSnippet(snippet string) {
 	}
 }
 
-// Snippets returns a deep copy of all the added snippets.
-func (spec *Specification) Snippets() map[string][]string {
-	result := make(map[string][]string, len(spec.snippets))
-	for k, v := range spec.snippets {
+// AddSunSnippet adds a new apparmor snippet for the snap-update-ns program.
```

Comment:
Since `snapName` is private to the implementation it cannot be set or manipulated. Perhaps just `AddSunSnippet adds a new apparmor snippet for the snap-update-ns program when operated in the context of the given snap`.

We could then either discard the map of snippets entirely (it is always one snap after all) or even make the snap name explicit here (and just cross-check it or ignore it). One thing to consider if we want to call AddSunSnippet just from backend code (for layouts) or also for the interface code (e.g. more strictly confined content interface)

Reply:
The content interface should be considered I think since it is a specialized interface that necessarily applies to all commands/hooks in the snap.